http/src/main/scala/org/apache/pekko/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -190,7 +202,7 @@ object Unmarshaller
 
     override def equals(that: Any): Boolean = that match {
       case that: UnsupportedContentTypeException =>
-        that.canEqual(this) && that.supported == this.supported && that.actualContentType == this.actualContentType
+        that.canEqual(this) && super.equals(this)
```

Comment:
This will point to `org.apache.pekko.http.javadsl.unmarshalling.Unmarshaller.UnsupportedContentTypeException` `equals` method which does the proper structural equality check.